expression/simple_rewriter.go

```diff
@@ -137,6 +137,11 @@ func (sr *simpleRewriter) Leave(originInNode ast.Node) (retNode ast.Node, ok boo
 		if v.Sel == nil {
 			sr.inToExpression(len(v.List), v.Not, &v.Type)
 		}
+	case *ast.ParamMarkerExpr:
+		tp := types.NewFieldType(mysql.TypeUnspecified)
+		types.DefaultParamTypeForValue(v.GetValue(), tp)
+		value := &Constant{Value: v.Datum, RetType: tp}
+		sr.push(value)
```

Comment:
I am confused about this newly added modification. Since we convert ParamMakerExpr to Constant here, what if we `execute` the prepared statement with another parameter? will it still use the previous constant instead of the parameter we provide?

Reply:
We rebuild the plan for every execution.
And each execution updates the value in the `ParamMakerExpr`.